src/com/palmergames/bukkit/towny/event/PlayerPreTransactionEvent.java

```diff
@@ -0,0 +1,47 @@
+package com.palmergames.bukkit.towny.event;
+
+import com.palmergames.bukkit.towny.object.Transaction;
+import org.bukkit.Bukkit;
+import org.bukkit.event.Cancellable;
+import org.bukkit.event.Event;
+import org.bukkit.event.HandlerList;
+
+public class PlayerPreTransactionEvent extends Event implements Cancellable {
```

Comment:
This should be renamed to TownyPreTransactionEvent, as it is not reliant on a player at all. (ie: it could be a town paying a nation, a nation paying the serveraccount.)